mm2src/mm2_metrics/src/mm_metrics.rs

```diff
@@ -0,0 +1,555 @@
+use common::executor::{spawn, Timer};
+#[cfg(not(target_arch = "wasm32"))] use common::log::error;
+use common::log::{LogArc, LogWeak};
+use itertools::Itertools;
+use metrics::{Key, Label};
+use mm2_err_handle::prelude::*;
+use serde_json::Value;
+use std::sync::{atomic::Ordering, Arc};
+use std::{collections::HashMap, slice::Iter};
+
+use crate::MmMetricsError;
+use crate::{common::log::Tag, MetricsOps, MmMetricsResult, MmRecorder};
+
+type MetricLabels = Vec<Label>;
+
+pub(crate) type MetricNameValueMap = HashMap<String, PreparedMetric>;
```

Comment:
This doesn't need to be public as it's only used inside `mm_metrics.rs`, also please remove the space above it.